### DISCUSSION TITLE
factorise the relation

### DISCUSSION THREAD

**Discussion by quentinalbertone** [Permify/permify#37] [Q&A]
Hello,

I would like to use permify for a project but also want to use it correctly since the beginning.

in this example:
```
entity user {} 

entity organization {
    relation member @user
    relation admin @user
} 

entity document {
    relation  owner  @user   
    relation  org    @organization      

    action view   = owner or org.member or org.admin
    action edit   = owner 
    action delete = owner
} 
```
is there a way to tell : `every member of the org (no matter their relation) can read the document`
something like `action view   = owner or org`.

Thanks

**Comment by EgeAytin** [comment]
Hi @quentinalbertone, since you already have a `member` relation in the organization, you can achieve the condition "every member of org can read any document" with the above schema. An additional thing to be aware of is, you need to assign every document to the organization so that members of it can view the documents that are assigned. So if we consider the above schema, when document X is created you need to create a relation tuple as `document:X#org@organization:1`. In that way, every member of organization 1 is now granted view access to document X.

**Reply by quentinalbertone** [reply]
thanks for your reply, I didn't explain myself very well.
I wanted to say every user from organisation can view documents and express it like this
```
entity user {} 

entity organization {
    relation member @user
    relation admin @user
} 

entity document {
    relation  owner  @user   
    relation  org    @organization      

    action view   = owner or org
    action edit   = owner 
    action delete = owner
} 
```

**Reply by EgeAytin** [reply]
Hi @quentinalbertone, unfortunately, you can not assing direct entity to an action. Also you don't need to do that, you already have a member relation in organization and as you stated we want to achiece every user in organization can view document. So, you can assign every user to a member in organization, (or to any new defined relation). The process is when any user assigned to an organization, you need tp write its relation tuple concurently to writeDB.
